data/convert.go

```diff
@@ -53,7 +53,10 @@ func NewWith(convert StructOptions, value interface{}) Value {
 	case reflect.String:
 		return String(v.String())
 	case reflect.Slice:
-		var slice []Value
+		if v.IsNil() {
+			return Null{}
```

Comment:
My app doesn't - in our system, this will never happen since we ensure that everything is initialized (non-nil).  I have no problem changing behavior across the board, but was trying to make the narrowest change possible to minimize impact on other consumers.

Also, json.Marshal does differentiate between nil and empty slices:

```
	var d = struct {
		Foo     []string
		Foofull []string
		Foonull []string
	}{
		Foo:     []string{},
		Foofull: []string{"hi"},
	}

	b, err := json.Marshal(d)
```
Result:
```
{"Foo":[],"Foofull":["hi"],"Foonull":null}
```